catroid/src/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -354,6 +354,10 @@ private Object interpretFunction(Functions function, Sprite sprite) {
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.toDegrees(Math.atan(doubleValueOfLeftChild));
 			case EXP:
 				return doubleValueOfLeftChild == null ? 0d : java.lang.Math.exp(doubleValueOfLeftChild);
+			case POWER:
+				return ((doubleValueOfLeftChild == null || doubleValueOfRightChild == null)
+						|| (doubleValueOfLeftChild < 0 && doubleValueOfRightChild < 1 && doubleValueOfRightChild > -1))
```

Comment:
what happens if the method would return NaN? (like it currently does with SQRT when a negative number is passed)

does it make sense to replicate all special cases/behaviours of all used methods of math library, or wouldnt it be better to refactor the whole switch case part of the method to instead of doing special case handling and right away returning method call result to write them some place and check for NaN after the switch case and then returning value?